modules/customize-certificates-add-service-serving-configmap.adoc

```diff
@@ -0,0 +1,48 @@
+// Module included in the following assemblies:
+//
+// * authentication/certificates/service-serving-certificate.adoc
+
+[id="add-service-certificate-configmap-{context}"]
+= Add a service certificate to a ConfigMap
+
+To secure communication to services that use a ConfigMap, 
+have the cluster generate a signed serving certificate and key pair into the
+ConfigMap. To do this, set the 
+`service.beta.openshift.io/inject-cabundle=true` 
+annotation on your ConfigMap. A certificate will be generated, which allows
+consumers to trust the generated `service-ca.crt` in their TLS client 
+configuration, allowing connections to services that use these certificates.
+
+.Procedure
+
+. Annotate the ConfigMap with `service.beta.openshift.io/inject-cabundle=true`.
++
+----
+$ oc annotate configmap <configmap-name> service.beta.openshift.io/inject-cabundle=true
+----
++
+[NOTE]
+====
+Explicitly referencing the `service-ca.crt` key in a volumeMount will 
+prevent a Pod from starting until the ConfigMap has been injected with
+the CA bundle.
+====
++
+For instance, to annotate the ConfigMap `foo` the following command would be
+used:
++
+----
+$ oc annotate configmap foo service.beta.openshift.io/inject-cabundle=true
+----
+
+. View the ConfigMap to ensure the certificate has been generated. This 
+appears as a `service-ca.crt` in the YAML output.
++
+----
+$ oc get configmap <configmap-name> -o yaml
+apiVersion: v1
+data:
+  service-ca.crt: |
+    -----BEGIN CERTIFICATE-----
+...
```

Comment:
We need to mention that the configmap object that you use for this purpose should not be shared with other configmap data. IE, if you have one configmap that has all your pod's config items in multiple keys, don't annotate it for injection because the other data will be deleted. So in this case use a second configmap exclusively for the service CA bundle.

Reply:
I've added an important block below the introduction that says this.